<?xml version="1.0" encoding="utf-8"?>

<launch>
  <arg name="camera_info" default="file://$(env HOME)/.ros/camera_info/0.yaml"/>
	<arg name="result_display" default="true"/>
	<arg name="result_autosize" default="true"/>
	<arg name="result_draw_markers" default="false" />
	<arg name="result_draw_markers_cube" default="false" />
	<arg name="result_draw_markers_axis" default="false" />

	<arg name="uid" default=""/>
  <arg name="video_namespace" default="/camera"/>
	<arg name="video_image_topic" default="image_rect_color"/>
	<arg name="video_info_topic" default="camera_info"/>
	<arg name="video_rectified" default="true" />

	<arg name="board_config" default="$(find ar_sys)/data/single/board.yml" />
  <arg name="board_frame_id" default="marker_link" />
  <arg name="marker_size" default="0.16" />
  <arg name="publish_tf" default="true" />

  <include file="$(find ar_sys)/launch/blackfly.launch" />

  <!-- undistort image -->
  <node ns="/camera" pkg="image_proc" type="image_proc" name="image_proc" >
    <remap from="image_raw" to="/camera/image_raw" />
    <remap from="camera_info" to="/camera/camera_info" />
  </node>


  <node ns="/" pkg="ar_sys" type="single_board" name="ar_single_board$(arg uid)">
		<remap from="/camera_info" to="$(arg video_namespace)/$(arg video_info_topic)" />
		<remap from="/image" to="$(arg video_namespace)/$(arg video_image_topic)" />
		<param name="image_is_rectified" type="bool" value="$(arg video_rectified)"/>
		<param name="board_config" type="string" value="$(arg board_config)"/>
		<param name="board_frame" type="string" value="/$(arg board_frame_id)" />

		<param name="marker_size" type="double" value="$(arg marker_size)"/>
		<param name="draw_markers" type="bool" value="$(arg result_draw_markers)" />
		<param name="draw_markers_cube" type="bool" value="$(arg result_draw_markers_cube)" />
		<param name="draw_markers_axis" type="bool" value="$(arg result_draw_markers_axis)" />
    <param name="publish_tf" value="$(arg publish_tf)" />
	</node>

	<node ns="/" pkg="topic_tools" type="relay" name="ar_single_board_relay$(arg uid)" args="/ar_single_board$(arg uid)/transform /arsys_single_board/transform" />

	<group if="$(arg result_display)">
		<node ns="/" pkg="image_view" type="image_view" name="image_view$(arg uid)" output="screen">
			<remap from="image" to="/ar_single_board$(arg uid)/result" />

			<param name="autosize" type="bool" value="$(arg result_autosize)" />
		</node>
	</group>

</launch>
